### ISSUE TITLE
[Feature Request] Distribute the application via F-Droid

### ISSUE THREAD

**Issue by Kratacoa** [acristescu/OnlineGo#74]
Given that the application is an open source front-end, it would be cool to have it distributed via an appropriate store.

**Comment by acristescu** [comment]
I have no qualms with distributing the app through F-Droid, although I suspect the number of people that would actually be installing it is quite low. However the problem comes from the fact the process seems to be quite involved... This is apparently just a [quick start](https://www.f-droid.org/en/docs/Submitting_to_F-Droid_Quick_Start_Guide/). To quote a living legend "ain't nobody got time for that" :)

As far as I could tell there is no easy way to deploy to F-Droid from Bitrise. If one such method could be found that takes no more than a couple of hours to implement I will be happy to do it though.

**Comment by bqv** [comment]
Registering interest, because ..."please!"

**Comment by malte-v** [comment]
I also think this app would be a great addition to F-Droid. However, we would first have to create a variant with all unfree dependencies removed, in particular Google's proprietary Play Services. Would you be fine with maintaining such a free/libre branch @acristescu?

**Comment by bqv** [comment]
Answer: no

**Comment by malte-v** [comment]
I'm sorry, what exactly is your role within this project @bqv?

**Comment by bqv** [comment]
Not involved, but I noticed you'd been ignored so I figured I'd let you know

**Comment by acristescu** [comment]
Sorry guys, I could swear I answered this somewhere, but obviously it was not here. @bqv might have seen that other message (wherever it is) and was just conveying the message.

TLDR; removing google play services is a hard task, one I do not have time to tackle. Even if somebody builds such a branch, I doubt I could maintain it.

Play services are used throughout the app:
- I think the smart polling will not work (since AFAIK WorkManager requires Play services - but investigation is needed).
- Crash reporting is dependent on firebase.
- Google Auth is dependent on play services.
- debug builds have stetho in them (which is Facebook)
- play billing is used for supporters

An alternative for each of these will need to be found and implemented.

If anybody wants to undertake something like this... be my guest.

**Comment by bqv** [comment]
Not quite the same, but since I'm maintaining a patchset, I've pushed my version to [https://skydroid.app](Skydroid) as **onlinego.skydroid.fron.io**.

I reccomend skydroid over f-droid anyway, because it's a better polished app, decentralized, and has f-droid bridges to have everything in the repo as a subset...

Happy to keep pushing updates to that, but I haven't made any modifications w.r.t "nonfree-ness" (yet?) so if that bothers you, steer clear

**Comment by bqv** [comment]
Updated to not have any of the stuff listed above, I think that makes it free now?

https://to.skydroid.app/onlinego.skydroid.fron.io

**Comment by m1karii** [comment]
@acristescu have you heard of / checked [IzzyOnDroid](https://apt.izzysoft.de/fdroid/index/info) repository - as to know whether it would be more suitable? from what i can tell quickly skimming through it they seem to have a lot less requirements for app inclusion; and as long as the only proprietary part is play services it should be ok (afaik)

**Comment by acristescu** [comment]
Sorry guys, it has been a frustrating experience trying to get the app on any of the OSS platform because I rely on Firebase crash reporting to keep an eye out on the health of the app. I cannot afford to pay testers so that's the only safety net I have. However, that causes all of the OSS repository maintainers to foam at the mouth and I've been laughed out of one too many such discussions.

If you want to install the app without Google Play, you actually can by using the releases section of this repository: https://github.com/acristescu/OnlineGo/releases
